subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -132,92 +24,117 @@ static void strncpy_nullterm(char *dst, const char *src, size_t maxlen)
 	}
 }
 
-int aws_fota_parse_UpdateJobExecution_rsp(char *update_rsp_document,
-					  size_t payload_len, char *status)
+int aws_fota_parse_UpdateJobExecution_rsp(const char *update_rsp_document,
+					  size_t payload_len, char *status_buf)
 {
-	if (update_rsp_document == NULL || status == NULL) {
+	if (update_rsp_document == NULL || status_buf == NULL) {
 		return -EINVAL;
 	}
 
-	struct update_rsp_obj rsp;
-
-	int ret = json_obj_parse(update_rsp_document, payload_len,
-			update_job_exec_stat_rsp_descr,
-			ARRAY_SIZE(update_job_exec_stat_rsp_descr), &rsp);
+	cJSON * update_response = cJSON_Parse(update_rsp_document);
 
-	if (ret < 0) {
-		return ret;
+	if (update_response == NULL) {
+		cJSON_Delete(update_response);
+		return -ENODATA;
 	}
 
-	/* Check if the status field(1st field) of the object has been parsed */
-	if (ret & BIT(UPDATE_JOB_EXECUTION_STATUS_DECODED_BIT)) {
-		strncpy_nullterm(status, rsp.status, STATUS_MAX_LEN);
+	cJSON * status = cJSON_GetObjectItemCaseSensitive(update_response,
+							  "status");
+	if (cJSON_IsString(status) && status->valuestring != NULL) {
+		strncpy_nullterm(status_buf, status->valuestring,
+				 STATUS_MAX_LEN);
 	} else {
+		cJSON_Delete(update_response);
 		return -ENODATA;
 	}
 
+	cJSON_Delete(update_response);
```

Comment:
Adding a `cleanup` tag here as well would make it consistent with below.

Reply:
Added